SODA/SodaClient.cs

```diff
@@ -478,5 +479,234 @@ internal TResult read<TResult>(Uri uri, SodaDataFormat dataFormat = SodaDataForm
+            Newtonsoft.Json.Linq.JObject payload = new Newtonsoft.Json.Linq.JObject();
+            Newtonsoft.Json.Linq.JObject source_type = new Newtonsoft.Json.Linq.JObject();
+            Newtonsoft.Json.Linq.JObject parse_option = new Newtonsoft.Json.Linq.JObject();
+            source_type["type"] = "upload";
+            source_type["filename"] = filename;
+            parse_option["parse_source"] = true;
+            payload["source_type"] = source_type;
+            payload["parse_options"] = parse_option;
+
+            var createSourceRequest = new SodaRequest(sourceUri, "POST", null, Username, password, SodaDataFormat.JSON, payload.ToString());
+            Result sourceOutput = createSourceRequest.ParseResponse<Result>();
+            string uploadDataPath = sourceOutput.Links["bytes"];
+            var uploadUri = SodaUri.ForUpload(Host, uploadDataPath);
+            Debug.WriteLine(uploadUri.ToString());
+            var fileUploadRequest = new SodaRequest(uploadUri, "POST", null, Username, password, dataFormat, data);
+            fileUploadRequest.SetDataType(SodaDataFormat.JSON);
+            Result result = fileUploadRequest.ParseResponse<Result>();
+            return new Source(result);
+        }
+
+        /// <summary>
+        /// Get the specified source data.
+        /// </summary>
+        /// <param name="source">The result of the Source creation</param>
+        /// <returns>A <see cref="Source"/>The updated Source object</returns>
+        public Source GetSource(Source source)
+        {
+            if (source == null)
+                throw new ArgumentException("Source required.", "source");
+            var sourceUri = SodaUri.ForSource(Host, source.Self());
+            var sourceUpdateResponse = new SodaRequest(sourceUri, "GET", null, Username, password, SodaDataFormat.JSON, "");
+            Result result = sourceUpdateResponse.ParseResponse<Result>();
+            return new Source(result);
+        }
+
+        /// <summary>
+        /// Create the InputSchema from the source.
+        /// </summary>
+        /// <param name="source">The result of the Source creation</param>
+        /// <returns>A <see cref="SchemaTransforms"/>SchemaTransforms object</returns>
+        public SchemaTransforms CreateInputSchema(Source source)
+        {
+            if (source == null)
+                throw new ArgumentException("Source required.", "source");
+            return new SchemaTransforms(source);
+        }
+
+        /// <summary>
+        /// Export the error rows (if present).
+        /// </summary>
+        /// <param name="filepath">The output file (csv)</param>
+        /// <param name="output">The specified transformed output</param>
+        public void ExportErrorRows(string filepath, AppliedTransform output)
+        {
+            if (String.IsNullOrEmpty(filepath))
+                throw new ArgumentException("Filepath must be specified.", "filepath");
+
+            if (output == null)
```

Comment:
might be able to us IsNullOrEmpty in these cases as well if you never expect an empty string to be valid